src/main/java/com/thirdcc/webapp/service/impl/DebtServiceImpl.java

```diff
@@ -73,6 +103,25 @@ public DebtDTO updateStatus(Long id, DebtStatus debtStatus) {
             add(DebtStatus.OPEN);
         }};
         return debtRepository.findAllByStatusIn(pageable, openDebtStatus)
-            .map(debtMapper::toDto);
+            .map(debtMapper::toDto)
+            .map(this::mapDebtDetails);
+    }
+    
+    private DebtDTO mapDebtDetails(DebtDTO debtDTO) {
+        Optional<EventAttendee> eventAttendee = eventAttendeeRepository.findById(debtDTO.getEventAttendeeId());
+        if(eventAttendee.isPresent()){
+            Optional<Event> event = eventRepository.findById(eventAttendee.get().getEventId());
+            if(event.isPresent()){
+                debtDTO.setEventName(event.get().getName());
+            }
+            Optional<User> dbUser = userRepository.findById(eventAttendee.get().getUserId());
+            if(dbUser.isPresent()){
+                User user = dbUser.get();
+                String lastName = (user.getLastName() != null ? " " + user.getLastName(): "");
+                String userName = user.getFirstName() + lastName;
+                debtDTO.setUserName(userName);
+            }
+        }
+        return debtDTO;
```

Comment:
This seems like something belongs to DebtDTO class. can pass in event, user object to hide the mapping logic maybe. Just opinion, not hard requirement